modules/ingester/instance.go

```diff
@@ -154,13 +172,49 @@ func newInstance(instanceID string, limiter *Limiter, overrides ingesterOverride
 	return i, nil
 }
 
-func (i *instance) PushBytesRequest(ctx context.Context, req *tempopb.PushBytesRequest) error {
+func (i *instance) PushBytesRequest(ctx context.Context, req *tempopb.PushBytesRequest) *tempopb.PushResponse {
+	var errors util.MultiError
+
+	maxLiveErrorTraces := make([]int32, 0)
+	traceTooLargeErrorTraces := make([]int32, 0)
+	totalTracesDiscarded := 0
+
 	for j := range req.Traces {
+		index := int32(j)
+
 		err := i.PushBytes(ctx, req.Ids[j].Slice, req.Traces[j].Slice)
 		if err != nil {
-			return err
+			errorMessage := err.Error()
+			totalTracesDiscarded++
+
+			if _, ok := err.(*maxLiveTracesError); ok {
+				maxLiveErrorTraces = append(maxLiveErrorTraces, index)
+				// only log one of each occurrence
+				if !strings.Contains(errorMessage, overrides.ErrorPrefixLiveTracesExceeded) {
+					errors.Add(err)
+				}
+			}
+
+			if _, ok := err.(*traceTooLargeError); ok {
+				traceTooLargeErrorTraces = append(traceTooLargeErrorTraces, index)
+				// only log one of each occurrence
+				if !strings.Contains(errorMessage, overrides.ErrorPrefixTraceTooLarge) {
+					errors.Add(err)
+				}
+			}
 		}
 	}
+
+	// at least one failure
+	if totalTracesDiscarded > 0 {
+		return &tempopb.PushResponse{
+			MaxLiveErrorTraces:       maxLiveErrorTraces,
+			TraceTooLargeErrorTraces: traceTooLargeErrorTraces,
+			Error:                    errors.Error(),
+		}
+	}
+
+	// no failure
```

Comment:
to reduce complexity it might make more sense to always return a value `tempopb.PushResponse`. even if there are no errors. in this case the slices will just be empty.